### PR TITLE
chore: karpenter core version bump and docgen

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
Fixes #N/A 

**Description**
Bumping core and metrics, looks like preview already contains cluster state.

**How was this change tested?**
NA
**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Bryce-Soghigian** [comment]
https://deploy-preview-5697--karpenter-docs-prod.netlify.app/docs/reference/metrics/#cluster-state-metrics

**Comment by Bryce-Soghigian** [comment]
We can /hold until the change gets released, just figured we should add docs pr for feature to be "complete"

File: website/content/en/docs/reference/metrics.md [hunk 1]
```diff
@@ -177,3 +177,10 @@ Maximum number of concurrent reconciles per controller
 ### `controller_runtime_active_workers`
```
Comment (jonathan-innis) [inline website/content/en/docs/reference/metrics.md:177]: This is auto-gened so you should be able to re-generate this metrics with `make docgen` after bumping the version of `kubernetes-sigs/karpenter` that this is referencing. Wanna just knock out two birds with one stone?

Comment (Bryce-Soghigian) [reply website/content/en/docs/reference/metrics.md:177]: I see so it docgens based on the karpenter core reference. 

Doing ` go get -u sigs.k8s.io/karpenter@release-v0.34.x` will give us the latest version of v0.34 but these changes aren't cherry picked there. 


Curious to learn how does the AWS process work. Do you break off a new branch and start developing against that version 0.35 or just continue to cherry pick into 0.34, then updating the provider?

Comment (Bryce-Soghigian) [reply website/content/en/docs/reference/metrics.md:177]: I bumped and did make docgen. THanks for sharing that command, seems preview already has the cluster state so happy to just bump to include update to your metrics.

**Comment by Bryce-Soghigian** [comment]
@jonathan-innis  thanks for sharing the commands! Seems you already fixed this in the past 3 days when i pushed the pr https://github.com/aws/karpenter-provider-aws/pull/5673/files, but happy to just bump and update the metrics

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**